aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move

```diff
@@ -0,0 +1,326 @@
+    struct State has key {
+        paused: bool,
+        denylist: SmartTable<address, bool>,
+    }
+
+    struct Approval has drop {
+        owner: address,
+        nonce: u64,
+        chain_id: u8,
+        spender: address,
+        amount: u64,
+    }
+
+    #[event]
+    struct Mint has drop, store {
+        minter: address,
+        to: address,
+        amount: u64,
+    }
+
+    #[event]
+    struct Burn has drop, store {
+        minter: address,
+        from: address,
+        store: Object<FungibleStore>,
+        amount: u64,
+    }
+
+    #[event]
+    struct Pause has drop, store {
+        pauser: address,
+        paused: bool,
+    }
+
+    #[event]
+    struct Denylist has drop, store {
+        denylister: address,
+        account: address,
+    }
+
+    #[view]
+    public fun usdk_address(): address {
+        object::create_object_address(&@stablecoin, ASSET_SYMBOL)
+    }
+
+    #[view]
+    public fun metadata(): Object<Metadata> {
+        object::address_to_object(usdk_address())
+    }
+
+    /// Called as part of deployment to initialize the stablecoin.
+    fun init_module(usdk_signer: &signer) {
+        let constructor_ref = &object::create_named_object(usdk_signer, ASSET_SYMBOL);
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            constructor_ref,
+            option::none(),
+            utf8(ASSET_SYMBOL), /* name */
+            utf8(ASSET_SYMBOL), /* symbol */
+            8, /* decimals */
+            utf8(b"http://example.com/favicon.ico"), /* icon */
+            utf8(b"http://example.com"), /* project */
+        );
+
+        // All resources created will be kept in the asset metadata object.
+        let metadata_object_signer = &object::generate_signer(constructor_ref);
+        move_to(metadata_object_signer, Roles {
+            master_minter: @master_minter,
+            minters: vector[@minter],
+            pauser: @pauser,
+            denylister: @denylister,
+        });
+
+        // Create mint/burn/transfer refs to allow creator to manage the stablecoin.
+        move_to(metadata_object_signer, Management {
+            extend_ref: object::generate_extend_ref(constructor_ref),
+            mint_ref: fungible_asset::generate_mint_ref(constructor_ref),
+            burn_ref: fungible_asset::generate_burn_ref(constructor_ref),
+            transfer_ref: fungible_asset::generate_transfer_ref(constructor_ref),
+        });
+
+        move_to(metadata_object_signer, State {
+            paused: false,
+            denylist: smart_table::new(),
+        });
+
+        // Override the deposit and withdraw functions which mean overriding transfer.
+        // This ensures all transfer will call withdraw and deposit functions in this module and perform the necessary
+        // checks.
+        let deposit = function_info::new_function_info(
+            usdk_signer,
+            string::utf8(b"usdk"),
+            string::utf8(b"deposit"),
```

Comment:
is this how devs "bind" their custom deposit implementation?

Reply:
Yep this and the register_dispatch_functions call below